### ISSUE TITLE
Rename `compute`, `reduce` and `scan` in the API

### ISSUE THREAD

**Issue by dguest** [dguest/lwtnn#16]
I originally called the method that runs the networks `compute`, but I didn't think very hard about it. Now there are a few other method names kicking around:
- `scan` for the RNN methods, which map a `MatrixXd` to another `MatrixXd`
- `reduce` for `RecursiveStack`, which maps a `MatrixXd` to a `VectorXd`. I named it this for consistency with the standard `reduce` function in most languages.

Maybe this isn't totally intuitive, and @makagan had the suggestion that we change the API to use `predict` for consistency with sklearn or Keras. If we're going to be changing the API for readability, I think we should do it consistently. So we should consider the following places:
- The `ILayer` derived classes and the `Stack` class.
- The `IRecurrent` derived classes and the `RecursiveStack` class.
- The `LightweightNeuralNetwork high-level wrapper on the`Stack` class. 
- I'm also going to write a `LightweightRNN` wrapper around `RecursiveStack` anyway, to take care of the variable naming, normalization, offsets, various sanity checks, etc. We could change both the `LightweightRNN` and `LightweightNeuralNetwork` classes to use `predict`, and keep the underlying api the same 

It's a small change, but it will have to happen consistently and obviously we'll only do it when everyone using the code is ready.

**Comment by dguest** [comment]
Closing this to clean up dead issues.
